src/coreclr/jit/emitxarch.cpp

```diff
@@ -15906,7 +16510,75 @@ BYTE* emitter::emitOutputRRR(BYTE* dst, instrDesc* id)
         dst += emitOutputByte(dst, (0xC0 | regCode));
     }
 
-    noway_assert(!id->idGCref());
+    // noway_assert(!id->idGCref());
+    if (id->idGCref())
+    {
+        assert(IsApxExtendedEvexInstruction(ins));
+        assert(id->idInsFmt() == IF_RWR_RRD_RRD);
+        switch (id->idIns())
```

Comment:
This looks like copy-and-paste code from `emitOutputRR`. Can we extract a function to handle both cases?